homeassistant/components/alarm_control_panel/elkm1.py

```diff
@@ -0,0 +1,165 @@
+"""
+Each ElkM1 area will be created as a separate alarm_control_panel in HASS.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/alarm_control_panel.elkm1/
+"""
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+import homeassistant.components.alarm_control_panel as alarm
+from homeassistant.const import (ATTR_ENTITY_ID, STATE_ALARM_ARMED_AWAY,
+                                 STATE_ALARM_ARMED_HOME,
+                                 STATE_ALARM_ARMED_NIGHT, STATE_ALARM_ARMING,
+                                 STATE_ALARM_DISARMED, STATE_ALARM_PENDING,
+                                 STATE_ALARM_TRIGGERED, STATE_UNKNOWN)
+
+from homeassistant.components.elkm1 import (DOMAIN, create_elk_devices,
+                                            ElkDeviceBase,
+                                            register_elk_service)
+from elkm1_lib.const import AlarmState, ArmedStatus, ArmLevel, ArmUpState
+
+DEPENDENCIES = [DOMAIN]
+
+STATE_ALARM_ARMED_VACATION = 'armed_vacation'
+STATE_ALARM_ARMED_HOME_INSTANT = 'armed_home_instant'
+STATE_ALARM_ARMED_NIGHT_INSTANT = 'armed_night_instant'
```

Comment:
The extra states aren't used yet because we couldn't use them on the old regular HASS UI, without them being added to polymer, etc. 

We've been trying to hammer out what additional states are needed so we can get them added to HASS properly.

See home-assistant/architecture#54 for more on that, if you're curious.